### PR TITLE
[FIX] crm: contact created without name

### PR DESCRIPTION
**Pull request by kmdi-odoo**
Steps to reproduce
===================
1. Create a new lead in CRM with only an email address.
2. Send a message to the email through the chatter. 
-> The contact is created but it has no name.

Issue
=================
The issue was in _get_customer_information() for CRM. The function uses
parse_contact_from_email which returns an empty name for a simple email.

After this commit
=================
Email will serve as a name for the contact created through the chatter in CRM.

Task-4129454

### CODE REVIEW

**Comment by mrsr-odoo** [comment]
Hello @kmdi-odoo 
It would be good to specify the issue in the commit that `parse_contact_from_email` returns an empty string for the name if simple email is passed


>The contact created by this way will have email as a name.

Should write like - `Email serves as the name`

**Comment by julien-banken** [comment]
Hello @kmdi-odoo and @mrsr-odoo :wave:,
Good job! Thank you for providing all the context and details about the task. It really helps me. I suppose that your lovely reviewer will ask you to write a small unit test to cover that specific corer case.

May I ask you to write a new test in `test_name_create_corner_cases` and add a single email address to the array? So, that we can check the the `parse_contact_from_email` function indeed returns the tuple `'', 'lenny.bar@gmail.com'` for `lenny.bar@gmail.com`. By reading the spec of that function, it's not really clear to me what it should return in that case. This will clarify things.

Then, you can also add a small test for the lead/contact creation to check that you indeed fallback to the email address when there is no name you can use :smiley:

**Comment by kmdi-odoo** [comment]
Hello @julien-banken :smiley:,
In the tests written for `test_name_create_corner_cases` the function `name_create` is used to create the partner. The function `name_create` has `self._rec_name: name or email_normalized` where `_rec_name` is `name` for `crm.lead` therefore in the test, email will be assigned as a name and hence it will not test that case.

**Comment by julien-banken** [comment]
Hello @kmdi-odoo,

Thank you for your follow up. I suggested you to add the "empty name corner case" to the test `test_name_create_corner_cases` to complete the existing test suite. It seems that we have a corner case here that is not covered by the test. So better to add it :smiley:

For your flow, you will have to adapt another existing test (if there already exists a test checking the lead/contact creation flow for instance) or create a new one. My message was maybe misleading as I didn't precise that the test for your flow should be written elsewhere.

**Review by julien-banken — APPROVED** [review]

**Comment by kmdi-odoo** [comment]
Hello @julien-banken :smiley:,
I have rectified the test to check the name as an email when the name is not provided. The earlier test was not failing even when my fix wasn't there :)

Thanks

File: addons/mail/tests/test_res_partner.py [hunk 1]
```diff
@@ -485,6 +485,7 @@ def test_name_create_corner_cases(self):
             'False',
             # (simili) void values
             '', ' ', False, None,
+            'lenny.bar@gmail.com',
```
Comment (tde-banana-odoo) [inline addons/mail/tests/test_res_partner.py:488]:
```suggestion
            # email only
            'lenny.bar@gmail.com',
```

File: addons/mail/tests/test_res_partner.py [hunk 2]
```diff
@@ -493,6 +494,7 @@ def test_name_create_corner_cases(self):
             ('False', False),
             # (simili) void values: always False
             ('', False), ('', False), ('', False), ('', False),
+            ('lenny.bar@gmail.com', 'lenny.bar@gmail.com')
```
Comment (tde-banana-odoo) [inline addons/mail/tests/test_res_partner.py:497]:
```suggestion
            # email only: email used as both name and email
            ('lenny.bar@gmail.com', 'lenny.bar@gmail.com')
```

**Review by tde-banana-odoo — COMMENTED** [review]
Linting comments, thanks for the fix :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**